specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/artifact.json

```diff
@@ -0,0 +1,1799 @@
+            "description": "The Azure subscription Id.",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of resource group in which the workspace is located.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "description": "The name of the workspace.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "artifact",
+            "in": "body",
+            "description": "The Artifact details.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Artifact"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Artifact is created successfully.",
+            "schema": {
+              "$ref": "#/definitions/Artifact"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/artifact/v2.0/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/artifacts/register": {
+      "post": {
+        "tags": [
+          "Artifact"
+        ],
+        "summary": "Create an Artifact for an existing data location.",
+        "description": "Create an Artifact for an existing dataPath.",
+        "operationId": "Artifacts_Register",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "subscriptionId",
+            "in": "path",
+            "description": "The Azure subscription Id.",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "description": "The name of resource group in which the workspace is located.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "workspaceName",
+            "in": "path",
+            "description": "The name of the workspace.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "artifact",
+            "in": "body",
+            "description": "The Artifact creation details.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Artifact"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
Operations that create resources should return 201/Created. Is there a specific reason as to why this operation doesn't?

Reply:
This issue is there with all our services. As the services are already being used by external customers, this is considered a breaking change by our service owners. Is this going to be a blocking issue?